### ISSUE TITLE
Cannot implement any custom IMediaSource implementations

### ISSUE THREAD

**Issue by SimpsOff** [martijn00/ExoPlayerXamarin#60]
IMediaSource missing the abstract methods `PrepareSource` and `CreatePeriod` which the corresponding java MediaSource class is expecting an implementation of. Because of this, there can be no implementations of the IMediaSource interface which compiles properly for the corresponding Java version. This is the error:

```
TrackExtractorMediaSource.java(4, 8): [null]  error: TrackExtractorMediaSource is not abstract and does not override abstract method createPeriod(MediaPeriodId,Allocator) in MediaSource
public class TrackExtractorMediaSource
```

I've tried adding in the exact method signature in my C# source without the interface having the method, and it just doesn't pick them up.

Now, it's obvious that these are missing because the `Metadata.xml` for `ExoPlayer.Core` has this:
```
<!-- Those methods mark the whole file invalid. Remove them for now -->
	<remove-node path="/api/package[@name='com.google.android.exoplayer2.source']/interface[@name='MediaSource']/method[@name='prepareSource']" />
	<remove-node path="/api/package[@name='com.google.android.exoplayer2.source']/interface[@name='MediaSource']/method[@name='createPeriod']" />
```
Now, I've been googling around and found this thread with the similar issue I belive: https://forums.xamarin.com/discussion/35713/xamarin-jar-binding-library-metadata-xml-tips-for-missing-abstract-method

So I tried pulling this project down and building myself (without any modifications), and i'm getting ~45 errors, so my environment must not be compatible (I'm using Visual Studio for Mac 7.3.3 build 12, Xamarin.Android 8.1.5.0). In the mean time i'm going to try and see if i can build locally, but i wanted to post the issue incase you had a solution (I haven't done much with Xamarin bindings...)

**Comment by martijn00** [comment]
The latest develop is not working yet because the bindings are not updated to 2.7.2. That needs to be done first before being able to fix this issue.

**Comment by martijn00** [comment]
This should be fixed in 2.8.1. Can you check?
